accepted/0000-ubuntu-sudo.md

```diff
@@ -0,0 +1,63 @@
+- Feature Name: automated ubuntu onboarding without need to manually configure sudo
+- Start Date: (2020-01-27)
+- RFC PR:
+
+# Summary
+[summary]: #summary
+
+As Ubuntu clients have no root users enabled by default, we tell the user to manually use visudo to enable a user for Python (i. e. for Salt)
+https://opensource.suse.com/doc-susemanager/suse-manager/develop/client-configuration/clients-ubuntu.html#_root_access
+
+This proposal tries to avoid the need to manually configure sudo before being able to bootstrap Ubuntu clients.
+
+# Motivation
+[motivation]: #motivation
+
+- Why are we doing this?
+
+Better user experience, make it easier to onboard Ubuntu minions
+
+- What use cases does it support?
+
+Onboarding Ubuntu minions
+
+- What is the expected outcome?
+
+Ubuntu clients should be as easy to onboard as any other client. This means it should be sufficient to just enter the root password in the SUSE Manager UI without also manually configuring the to-be-onboarded Ubuntu minion.
+
+# Detailed design
+[design]: #detailed-design
+
+When onboarding a minion via ssh, SUSE Manager asks for a user and password with sufficient privileges. On non-Ubuntu systems this typically is the root user. On Ubuntu systems, there typically is just an ordinary user, but this user can do almost anything via sudo. This user is referred to as "ubuntuser" in this document. Unfortunately it is necessary to supply the user's password to sudo. So currently the SUSE Manager documentation asks the admin to enable some special commands in sudo to be executed even without providing the password. The goal is to do this modification directly from SUSE Manager, so there is no need to manually prepare the system for onboarding.
```

Comment:
Do I get it correctly that you are addressing the onboarding from the Web UI only in this RFC?

(ie. onboarding via manual execution of a bootstrap script and onboarding via key acceptance do not really change)

Reply:
Yes, this is only about bootstrapping via the UI. If the user is logged into the minion and manually executes the bootstrap script, there is not much to win anyway as he can also give herself the needed privileges.